substrafl/algorithms/pytorch/torch_newton_raphson_algo.py

```diff
@@ -249,30 +249,6 @@ def _local_train(
 
             self._update_gradients_and_hessian(loss, current_batch_size)
 
-    def _local_predict(self, predict_dataset: torch.utils.data.Dataset, predictions_path):
-        """Executes the following operations:
-
-            * Create the torch dataloader using the batch size given at the ``__init__`` of the class
-            * Sets the model to `eval` mode
-            * Returns the predictions
-
-        Args:
-            predict_dataset (torch.utils.data.Dataset): predict_dataset build from the x returned by the opener.
-        """
-        predict_loader = torch.utils.data.DataLoader(predict_dataset, batch_size=self._batch_size)
```

Comment:
What you can do instead to remove the bug, is to write `predict_loader = torch.utils.data.DataLoader(predict_dataset, batch_size=self._batch_size or len(predict_dataset))`